docs/guides/rag.md

```diff
@@ -0,0 +1,126 @@
+---
+tags:
+  - RAG
+---
+# RAG Implementation Guide
+
+![rag](../images/rag.png)
+
+## Preparation Phase
+
+### 1. Define Business Requirements
+
+- Clearly outline the specific use cases and goals for the RAG system
+- Identify key performance indicators (KPIs) to measure success
+
+### 2. Gather Test Documents
+
+- **Pertinence** - documents must meet the business requirements
+- **Representativeness** - documents should be representative of all the types of documents that your solution will use
+- **Physical document quality** - documents need to be in a usable shape, e.g. clear document scans
+- **Document content quality** - documents must have high content quality, e.g. there should not be misspellings or grammatical errors
+
+Hints:
+
+- Remember to reduct PII from real documents
+- Have at least two documents for each document type
+- If using synthetic documents, ensure they are as close to real data as possible
+- You can use LLMs to help evaluate the document quality
+
+### 3. Gather Test Queries
+
+- Create a diverse set of queries covering various use cases
+- Include edge cases and potential challenging scenarios
+
+## Chunking Phase
+
+### 1. Perform Document Analysis
+
+- What information should be ignored or excluded?
+- What information should be captured in chunks?
+- How should the document be chunked? (e.g. by sentence or fixed size with overlap)
+
+### 2. Choose Chunking Method
```

Comment:
Maybe a controversial one but could we add manual chunking here? I.e. a human curates the document and chunks?